src/MiddlewareFactory.php

```diff
@@ -121,4 +125,114 @@ private function isArrayDefinition(mixed $definition): bool
 
         return is_subclass_of((string) ($definition['class'] ?? ''), MiddlewareInterface::class);
     }
+
+    /**
+     * @param array{0:class-string, 1:string}|Closure $callable
+     */
+    private function wrap(array|Closure $callable): MiddlewareInterface
+    {
+        if (is_array($callable)) {
+            return $this->createActionWrapper($callable[0], $callable[1]);
+        }
+
+        return $this->createCallableWrapper($callable);
+    }
+
+    private function createCallableWrapper(callable $callback): MiddlewareInterface
+    {
+        return new class ($callback, $this->container, $this->parametersResolver) implements MiddlewareInterface {
+            private $callback;
+
+            public function __construct(
+                callable $callback,
+                private ContainerInterface $container,
+                private ParametersResolverInterface $parametersResolver
```

Comment:
Maybe make parameter resolver is nullable. It's allow don't use refleaction for simple cases.

Reply:
Done.